torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -379,11 +380,40 @@ class TensorExprFuser {
       }
 
       Node* fusion_group = createFusionGroup(n);
-      fusion_groups.push_back(fusion_group);
-      it = fusion_group->reverseIterator();
+      debugDumpFusionGroup("Fusion group constructed: ", fusion_group);
+
+      // Try merging the just created fusion group into the previous one.
+      // If it did not work, then put the previous fusion group into
+      // fusion_groups vector - we will not touch it anymore in this loop.
+      // If merging suceeded, save the merged group as the "previous" fusion
+      // group so that we can try to merge the next one into it.
+      if (prev_fusion_group) {
+        debugDumpFusionGroup(
+            "Trying to merge into the previous fusion group: ",
+            prev_fusion_group);
+        if (canMerge(prev_fusion_group, fusion_group)) {
```

Comment:
tryMerge can fail even when `canMerge` can fail, which is a little confusing. https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/passes/tensorexpr_fuser.cpp#L443

Can we inline `canMerge` into tryMerge, and then call tryMerge here instead of `canMerge` ? (these are the only two callsites)